src/Service.Tests/SqlTests/GraphQLMutationTests/GraphQLMutationTestBase.cs

```diff
@@ -677,6 +677,185 @@ public async Task DeleteMutationWithVariablesAndMappings(string dbQuery, string
+                    id
+                    title    
+                }
+                }
+            ";
+
+            string graphQLMutation2 = @"
+                mutation {
+                    createComic (item: { id : 5001, categoryName: ""Fantasy"", title: ""Lord of the Rings""}){
+                    id
+                    title    
+                }
+                }
+            ";
+
+            Task<JsonElement> responeTask1 = ExecuteGraphQLRequestAsync(graphQLMutation1, graphQLMutationName, isAuthenticated: true);
+            Task<JsonElement> responseTask2 = ExecuteGraphQLRequestAsync(graphQLMutation2, graphQLMutationName, isAuthenticated: true);
+
+            JsonElement response1 = await responeTask1;
+            JsonElement response2 = await responseTask2;
+
+            string responseString1 = response1.ToString();
+            string responseString2 = response2.ToString();
+            string expectedStatusCode = $"{DataApiBuilderException.SubStatusCodes.DatabaseOperationFailed}";
+
+            // It is not possible to know beforehand which mutation created the new item. So, validations
+            // are performed for the cases where either mutation could have succeeded. In each case,
+            // one of the mutation's reponse will contain a valid repsonse and the other mutation's
+            // response would contain DatabaseOperationFailed sub-status code as it would've failed at
+            // the database layer due to primary key violation constraint.
+            if (responseString1.Contains($"\"code\":\"{expectedStatusCode}\""))
+            {
+                Assert.AreEqual("{\"id\":5001,\"title\":\"Lord of the Rings\"}", responseString2);
+            }
+            else if (responseString2.Contains($"\"code\":\"{expectedStatusCode}\""))
+            {
+                Assert.AreEqual("{\"id\":5001,\"title\":\"Harry Potter\"}", responseString1);
+            }
+            else
+            {
+                Assert.Fail("Unexpected error. Atleast one of the mutations should've succeeded");
+            }
+        }
+
+        /// <summary>
+        /// Performs concurrent delete mutations on the same item and validates that only one of the
+        /// requests is successful.
+        /// </summary>
+        [TestMethod]
+        public async Task TestParallelDeleteMutations()
+        {
+            string graphQLMutationName = "deletebook";
+
+            string graphQLMutation1 = @"
+                mutation {
+                  deletebook (id: 1){
+                    id
+                    title
+                  }
+                }
+            ";
+
+            Task<JsonElement> responeTask1 = ExecuteGraphQLRequestAsync(graphQLMutation1, graphQLMutationName, isAuthenticated: true);
+            Task<JsonElement> responseTask2 = ExecuteGraphQLRequestAsync(graphQLMutation1, graphQLMutationName, isAuthenticated: true);
+
+            JsonElement response1 = await responeTask1;
```

Comment:
Same question: how do you guarantee the 2 deletes are triggered concurrently? is it by relying on the async operation?

Reply:
Right. The requests are triggered concurrently because of the usage of async. 

L834: Using the await clause, forces the test to block until the async task (responseTask1) completes execution and returns the result. 

Similarly, L835 blocks the test execution until async task (responseTask2) completes execution and returns the result.

When response1.ToString()/response2.ToString() methods are called, both the response1 and response2 would have the results computed.